### DISCUSSION TITLE
How is Hyperparameter Optimization Done?

### DISCUSSION THREAD

**Discussion by BradKML** [alan-turing-institute/sktime#3613] [Q&A]
In the original AutoML library, they seemed to create a hyperparameter tuning system for all possible TS algorithms, which is slightly illegible https://github.com/winedarksea/AutoTS/blob/master/autots/models/sklearn.py
Can Sktime auto-implement defaults for each forecaster e.g. `UnobservedComponents` equivalent for `AutoETS` or `AutoARIMA`? If so, what are the avenues of which it can be done?

See also https://github.com/winedarksea/AutoTS/discussions/141

**Comment by RahulDas-dev** [comment]
AutoTS uses Genetic Algorithm, Other Approach would be Bayesian hyperparameter optimization

**Comment by fkiraly** [comment]
> Can Sktime auto-implement defaults for each forecaster e.g. UnobservedComponents equivalent for AutoETS or AutoARIMA? If so, what are the avenues of which it can be done?

If you want to implement special parameter sets, you can extend or override the `get_test_params` API, this allows you to attach parameter sets to each estimator. Currently, it is used for testing only, but you can also define "favourite defaults" that way.
